src/eslint-action.ts

```diff
@@ -108,15 +112,38 @@ async function run(): Promise<void> {
       });
       const report = lint(files);
       const payload = processReport(report);
-
-      await oktokit.checks.update({
-        owner: OWNER,
-        repo: REPO,
-        completed_at: new Date().toISOString(),
-        status: 'completed',
-        check_run_id: checkId,
-        ...payload,
-      });
+      const maxChunk = 50;
+
+      if (payload?.output?.annotations.length > maxChunk) {
+        const chunks = Math.ceil(payload.output.annotations.length / maxChunk);
+        core.info(`There were ${payload.output.annotations.length} annotations, splitting into ${chunks} requests`);
+        for (let index = 0; index < chunks; index++) {
+          const startIndex = index * maxChunk;
+          core.info(`Applying annotations ${startIndex} to ${startIndex + maxChunk}...`);
+          const returnValue = await oktokit.checks.update({
+            owner: OWNER,
+            repo: REPO,
+            completed_at: new Date().toISOString(),
+            status: 'completed',
+            check_run_id: checkId,
+            conclusion: payload.conclusion,
+            output: {
+              ...payload.output,
+              annotations: payload.output.annotations.slice(startIndex, startIndex + maxChunk),
+            },
+          });
```

Comment:
Should this be broken down a bit? Adding the annotations to the check and then updating it with a complete status?